skiko/src/nativeJsMain/cpp/Font.cc

```diff
@@ -24,381 +24,277 @@ SKIKO_EXPORT KNativePointer org_jetbrains_skia_Font__1nMakeDefault
+    res[0] = bounds.left();
+    res[1] = bounds.top();
+    res[2] = bounds.right();
+    res[3] = bounds.bottom();
 }
 
-#if 0
-SKIKO_EXPORT KFloat org_jetbrains_skia_Font__1nMeasureTextWidth
-  (KNativePointer ptr, KInteropPointer str, KNativePointer paintPtr) {
-    SkFont* instance = reinterpret_cast<SkFont*>((ptr));
-    SkPaint* paint = reinterpret_cast<SkPaint*>((paintPtr));
-    jsize len = env->GetStringLength(str);
-    const KChar* chars = env->GetStringCritical(str, nullptr);
-    float width = instance->measureText(chars, len * sizeof(KChar), SkTextEncoding::kUTF16, nullptr, paint);
-    env->ReleaseStringCritical(str, chars);
-    return width;
-}
-#endif
-
 
-
-SKIKO_EXPORT KFloat* org_jetbrains_skia_Font__1nGetWidths
-  (KNativePointer ptr, KShort* glyphsArr) {
-    TODO("implement org_jetbrains_skia_Font__1nGetWidths");
-}
-
-#if 0
-SKIKO_EXPORT KFloat* org_jetbrains_skia_Font__1nGetWidths
-  (KNativePointer ptr, KShort* glyphsArr) {
-    SkFont* instance = reinterpret_cast<SkFont*>((ptr));
-    int count = env->GetArrayLength(glyphsArr);
-    std::vector<KFloat> widths(count);
-    KShort* glyphs = env->GetShortArrayElements(glyphsArr, nullptr);
-    instance->getWidths(reinterpret_cast<SkGlyphID*>(glyphs), count, widths.data());
-    env->ReleaseShortArrayElements(glyphsArr, glyphs, 0);
-    return javaFloatArray(env, widths);
+SKIKO_EXPORT KFloat org_jetbrains_skia_Font__1nMeasureTextWidth
+  (KNativePointer ptr, char* str, KInt len, KNativePointer paintPtr) {
+    SkFont* instance = reinterpret_cast<SkFont*>(ptr);
+    SkPaint* paint = reinterpret_cast<SkPaint*>(paintPtr);
+    return instance->measureText(str, len * sizeof(KChar), SkTextEncoding::kUTF16, nullptr, paint);
 }
-#endif
-
 
-
-SKIKO_EXPORT KInteropPointerArray org_jetbrains_skia_Font__1nGetBounds
-  (KNativePointer ptr, KShort* glyphsArr, KNativePointer paintPtr) {
-    TODO("implement org_jetbrains_skia_Font__1nGetBounds");
+SKIKO_EXPORT void org_jetbrains_skia_Font__1nGetWidths
+  (KNativePointer ptr, KShort* glyphs, KInt count, KFloat* widths) {
+    SkFont* instance = reinterpret_cast<SkFont*>(ptr);
+    instance->getWidths(reinterpret_cast<SkGlyphID*>(glyphs), count, widths);
 }
 
-#if 0
-SKIKO_EXPORT KInteropPointerArray org_jetbrains_skia_Font__1nGetBounds
-  (KNativePointer ptr, KShort* glyphsArr, KNativePointer paintPtr) {
-    SkFont* instance = reinterpret_cast<SkFont*>((ptr));
-    SkPaint* paint = reinterpret_cast<SkPaint*>((paintPtr));
-    int count = env->GetArrayLength(glyphsArr);
+SKIKO_EXPORT void org_jetbrains_skia_Font__1nGetBounds
+  (KNativePointer ptr, KShort* glyphs, KInt count, KNativePointer paintPtr, KFloat* res) {
+    SkFont* instance = reinterpret_cast<SkFont*>(ptr);
+    SkPaint* paint = reinterpret_cast<SkPaint*>(paintPtr);
     std::vector<SkRect> bounds(count);
-    KShort* glyphs = env->GetShortArrayElements(glyphsArr, nullptr);
     instance->getBounds(reinterpret_cast<SkGlyphID*>(glyphs), count, bounds.data(), paint);
```

Comment:
Since SkRect is represented as 4 floats(ltrb), you can write directly to res as `SkRect*`.